README.md

```diff
@@ -32,7 +34,143 @@ For good first issues regarding contributing, see [issues labeled with open sour
+```
+
+To work with the frontend, you shouldn't need any further variable than the ones already provided.
+Follow the instructions inside `.envrc` if you need to set up a mock API endpoint.
+
+Finally, you need to read the variables from `.envrc`:
+
+```shell
+source .envrc
+```
+
+This command will also make sure that you are using the right Node version for this project.
+
+## Available NPM tasks (frontend)
+
+- `frontend-main-start` runs the React application of Oiretutka
+- `frontend-main-build` creates a build for the React application
+- `frontend-embed-v1-start` runs the jQuery embed of Oiretutka
+- `frontend-embed-v1-build` creates a build for the jQuery embed
+
+## Wrap up
+
+For the following times you will work on the project, you will need to
+
+- enter the project folder
+- run `source .env`, which will automatically set up the right node version (assuming you are using nvm)
+- `npm run frontend-main-start` or `npm run frontend-embed-v1-start` according to which part of the project you want to run locally
+
+## Developing the stand alone (React) app
+
+## Developing the embedded form (with vanilla JS, jQuery and TypeScript)
+
+As said earlier in this document, the crowdsourcing happens by asking questions in an embed that is placed on news media pages or articles.
+
+Performance wise, it didn't seem like a good idea to develop the embedded form with a modern JS framework and have all the users who are opening the news articles download extra data, when the form can be implemented using basic HTML and very light-weight (vanilla) JavaScript and JS libraries (jQuery). The JavaScript is typed with TypeScript.
+
+If you have worked previously with modern JS frameworks, these technologies might feel a bit old-fashioned and foreign. If you haven't used vanilla JavaScript in a while or haven't noticed you know vanilla JS, [read and bookmark this cheatsheet](https://htmldom.dev/), [practice JavaScript interactively online](https://learnjavascript.online/) or just [browse extensive JS documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript). If your jQuery skills are rusty, or you haven't worked with jQuery before, [take a look at jQuery basics](http://jqfundamentals.com/chapter/jquery-basics).
+
+If you haven't worked with TypeScript before, take a look at [the TypeScript Handbook](https://www.typescriptlang.org/docs/handbook/basic-types.html). Mostly TypeScript means that all function parameters and outputs have to be typed. If [the basic types](https://www.typescriptlang.org/docs/handbook/basic-types.html) aren't enough, take a look at [jQuery types](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/682ebb5fcda0d6d68bda36862171cd1f116121b2/types/jquery) too.
+
+Also, as the form is emdedded through an <iframe> element, it's a good idea to read [the ultimate guide to iframes](https://blog.logrocket.com/the-ultimate-guide-to-iframes/), it'll give you more context about how the media sites will use this form.
```

Comment:
Beautiful stuff here, @sjarva 😎 

![](https://media0.giphy.com/media/KI9oNS4JBemyI/giphy.gif?cid=5a38a5a2c7eec342e3efb0fb9cbb60a2b906a02b965d945d&rid=giphy.gif)

Reply:
Aww, thanks for the praise @mahdava 😍 Made my day ❤️ ❤️ ❤️